Sudoku/Controller/GameController.cs

```diff
@@ -59,11 +62,39 @@ public void RunGame(AbstractBoard board)
                     case ConsoleKey.Enter:
                         _game.Select.Execute();
                         break;
+                    case ConsoleKey.Spacebar:
+                        _game.Solver.SolveBoard((_game.Board.SudokuBoard as SudokuBoard)!);
+                        gameOver = true;
+                        break;
                 }
                 
                 ReDraw();
+                
+                if (gameOver) break;
             }
+            if (gameOver) break;
         } while (Console.ReadKey(true).Key != ConsoleKey.Escape);
+        
+        return RequestNewGame();
+    }
+
+    private bool RequestNewGame()
+    {
+        ConsoleKeyInfo keyPressed;
+        do
+        {
+            _boardView.StartNewGameMessage();
+            keyPressed = Console.ReadKey(true);
+
+            if (keyPressed.Key == ConsoleKey.Y)
+                return true;
+
+            if (keyPressed.Key == ConsoleKey.N)
+            {
+                _boardView.EndGameMessage();
+                return false;
+            }
+        } while (true);
     }
 
     public void ReDraw(List<ISimpleViewMessage>? pre = null, List<ISimpleViewMessage>? post = null)
```

Comment:
we hebben vrij veel nullable/ == null in ons programma en martijn gaat daar vragen over stellen